src/pal/tests/palsuite/file_io/FILECanonicalizePath/FILECanonicalizePath.c

```diff
@@ -0,0 +1,94 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+//
+
+/*=====================================================================
+**
+** Source:  FILECanonicalizePath.c (test 1)
+**
+** Purpose: Tests the PAL implementation of the FILECanonicalizePath function.
+**
+**
+**===================================================================*/
+
+#include <palsuite.h>
+
+#define TEST_ARR_SIZE 10
+extern void FILECanonicalizePath(LPSTR lpUnixPath);
+
+int __cdecl main(int argc, char *argv[])
+{
+	if (PAL_Initialize(argc,argv) != 0)
+    {
+        return FAIL;
+    }
+
+    char paths[TEST_ARR_SIZE] = {};
+
+    // Case 1: // transforms to /
+    strcpy(paths, "//");
+    FILECanonicalizePath(paths);
+    if (strcmp(paths, "/") != 0)
+    {
+    	Fail("FILECanonicalizePath error: // did not transform to /, translated to: %s", paths);
+    }
+    memset(paths, '\0', TEST_ARR_SIZE);
```

Comment:
All of these cases appear to follow the same pattern.  Consider adding a simple Test method that takes the input and expected paths; then each of these cases is just a one-liner to call that function with the path to canonicalize and the expected result to compare against.